Update description of config_automation and screenshots (#15)
ottrproject/ottrproject.github.io: customize-robots.Rmd

1) One that is run during pull requests. It is set by:
```
url-checker: true
```

GitHub Actions runs a check on all the URLs upon creating a pull request to the `main` branch.
If it fails, you can click on the output comment on your pull request that says "Click here to download detailed error reports". This will download a zip file containing a list of the broken URLs it found.

> If you don't set the additional error threshold option in the config_automation file, the check summary may still report that the URL check failed, even if 0 errors were found.

Reviewer: ```suggestion
It is necessary to set the error threshold in the config_automation file (see below); otherwise, the check summary may still report that the URL check failed, even if no errors were found.
```

Reviewer: trying to make this a bit more clear

Author: Much clearer!! I like it

In the `config_automation.yml` file it is set by:
```
spell-check: true
```
Github Actions will automatically run a spell check on all Rmds and mds whenever a pull request to the `main` branch is filed.

> The error threshold option controls how many errors are allowed before the check will fail.

Reviewer: ```suggestion
Just like the URL error threshold, the spelling error thershold option controls how many errors are allowed before the check will fail.
```

```
spell_error_min: 0
```

> You'll need to resolve those before being able to merge your pull request. Errors are displayed in much the same way as PR URL check errors with a summary of how many errors are found. Click on the output comment on your pull request that says "Click here to download detailed error reports". This will download a zip file containing a list of the spelling errors it found.

Reviewer: ```suggestion
You'll need to resolve those before being able to merge your pull request. Errors are displayed in much the same way as URL check errors with a summary of how many errors are found. Click on the output comment on your pull request that says "Click here to download detailed error reports". This will download a zip file containing a list of the spelling errors it found.
```

Some of these errors may be things that the spell check doesn't recognize for example: `ITCR` or `DaSL`.
If it's a 'word' the spell check should recognize, you'll need to add this to the dictionary.

Go to the `resources/dictionary.txt` file. Open the file and add the new 'word' to its appropriate place (the words are in alphabetical order).
After committing the changes to `resources/dictionary.txt` on your branch, the spell check status check should be successfully passed.

> The PR comment included below shows an example of a check that passed (spell check), and a check that failed (the URL check) with the option to download detailed error report(s).

Reviewer: ```suggestion
The pull request comment included below shows an example of a check that passed (spell check), and a check that failed (the URL check) with the option to download detailed error report(s).
```

### Style code

In the `config_automation.yml` file it is set by:
```
style-code: true
```
The `styler` package will style R code in all Rmds. Style changes will automatically be committed back to your branch.

### Docker testing

In the config_automation.yml file it looks like:
```
docker-test: false
```
By default it is set to `false` which means it won't run automatically unless you change this to `true`.
This is only relevant if you have [your own Docker image you are managing for your course](https://www.ottrproject.org/customize-docker.html). If changes are made to Docker-relevant files: `Dockerfile`, `install_github.R`, or `github_package_list.tsv`, this will test re-build the Docker image. If it is successfully built, then it makes sense to merge it to `main` but the Docker Image will not be pushed to `Dockerhub` automatically. [Follow these instructions to push your Docker image to Dockerhub](https://www.ottrproject.org/customize-docker.html#Testing_a_modified_Docker_image).

<br>

## Rendering actions

Upon merging changes to any `Rmd` or `assets/` folder to `main`, the course material will be automatically re-rendered.

> For rendering a website version of the course (or your website), rather than `true` or `false`, the options are `rmd`, `rmd_web`, `quarto` or `quarto_web`.

Reviewer: ```suggestion
For rendering a website version of the course (or your website), rather than `true` or `false`, the options are `rmd`, `rmd_web`, `quarto` or `quarto_web`.

A course has chapters like a book, where as a website version, will appear more like a typical website.
```

Reviewer: hmmm I feel like we might need additional guidance about how these differ

  * If rendering a course with Rmarkdown files, use `rmd`.

>   * If rendering a course with quarto files, use `quarto`.

Reviewer: ```suggestion
  * If rendering a course with [quarto files](https://quarto.org/), use `quarto`.
```

>   * If rendering a website with Rmarkdown files, use `rmd_web`.

Reviewer: ```suggestion
  * If rendering a website with [R Markdown files](https://rmarkdown.rstudio.com/), use `rmd_web`.
```

>   * If rendering a website with quarto files, use `quarto_web`.

Reviewer: ```suggestion
  * If rendering a website with [quarto files](https://quarto.org/), use `quarto_web`.
```

```
render-website: rmd
```

> By default, all rendering steps will be run. But depending on the needs of your course, you can turn these on and off by going to the `config_automation.yml` file and switching options to `true` or `false`.

Reviewer: ```suggestion
By default, all rendering steps will be run. But depending on the needs of your course or website, you can turn these on and off by going to the `config_automation.yml` file and switching options to `true` or `false`.
```

In the `config_automation.yml` file it is set by:
```
spell-check: true
```
Github Actions will automatically run a spell check on all Rmds and mds whenever a pull request to the `main` branch is filed.

> Just like the URL error threshold, the spelling error thershold option controls how many errors are allowed before the check will fail.

Author: ```suggestion
Just like the URL error threshold, the spelling error threshold option controls how many errors are allowed before the check will fail.
```

```
spell_error_min: 0
```

> You'll need to resolve those before being able to merge your pull request. Errors are displayed in much the same way as URL check errors with a summary of how many errors are found. Click on the output comment on your pull request that says "Click here to download detailed error reports". This will download a zip file containing a list of the spelling errors it found.

Author: I believe you're correct that it's more of a strongly recommend -- someone could set up repo settings such that all checks/workflows have to pass before they can merge, but they can still merge by clicking a bypass button. @carriewright11 do we want to soften the description at all since it's more of a strongly recommended than a requirement for default settings?

> You'll need to resolve those before being able to merge your pull request. Errors are displayed in much the same way as URL check errors with a summary of how many errors are found. Click on the output comment on your pull request that says "Click here to download detailed error reports". This will download a zip file containing a list of the spelling errors it found.

Reviewer: LGTM!

> You'll need to resolve those before being able to merge your pull request. Errors are displayed in much the same way as URL check errors with a summary of how many errors are found. Click on the output comment on your pull request that says "Click here to download detailed error reports". This will download a zip file containing a list of the spelling errors it found.

Reviewer: might need to check if the URL checker documentation says something similar about not being able to merge, but maybe you already checked that

> You'll need to resolve those before being able to merge your pull request. Errors are displayed in much the same way as URL check errors with a summary of how many errors are found. Click on the output comment on your pull request that says "Click here to download detailed error reports". This will download a zip file containing a list of the spelling errors it found.

Author: I searched all occurrences of the word "need", but I'll double check now. If it's good, I'll go ahead and commit and then merge -- thanks!!

Some of these errors may be things that the spell check doesn't recognize for example: `ITCR` or `DaSL`.
If it's a 'word' the spell check should recognize, you'll need to add this to the dictionary.

Go to the `resources/dictionary.txt` file. Open the file and add the new 'word' to its appropriate place (the words are in alphabetical order).
After committing the changes to `resources/dictionary.txt` on your branch, the spell check status check should be successfully passed.

The pull request comment included below shows an example of a check that passed (spell check), and a check that failed (the URL check) with the option to download detailed error report(s).

![](https://raw.githubusercontent.com/ottrproject/ottrproject.github.io/kweav-actionsDocsUpdate/resources/screenshots/check_results.png)

When all checks pass, the option to download error reports is no longer included, instead displaying a message that the URL and spell checks passed.

> ![](https://raw.githubusercontent.com/ottrproject/ottrproject.github.io/kweav-actionsDocsUpdate/resources/screenshots/check_results_pass.png)

Author: ```suggestion
![](https://raw.githubusercontent.com/ottrproject/ottrproject.github.io/main/resources/screenshots/check_results_pass.png)
```
